synthtool/gcp/partials.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import yaml
+from pathlib import Path
+from typing import Dict, List
+
+# these are the default locations to look up
+_DEFAULT_PARTIAL_FILES = [".readme-partials.yml", ".readme-partials.yaml"]
+
+
+def load_partials(files: List[str] = _DEFAULT_PARTIAL_FILES) -> Dict:
+    """
+    hand-crafted artisinal markdown can be provided in a .readme-partials.yml.
+    The following fields are currently supported:
```

Comment:
I like this refactor 👍 

It might be worth running the branch against one Node.js repo that uses partials, just to make sure it behaves as expected.

Reply:
Ran this against nodejs-storage and it succeeded with no changes to the README.md